newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -145,6 +154,26 @@ class ElectronMainMenu extends React.Component<Props, {||}> {
     }
   }
 
+  getRecentFiles = () => {
+    const recentFiles = localStorage.getItem('gd-recent-files');
+    if (recentFiles) {
+      const parsedData = JSON.parse(recentFiles);
+      const menuArray = [];
+      for (const prop in parsedData) {
```

Comment:
Ideally we want to avoid:
- having direct usage of localStorage inside our components (this might impact performance as it's a synchronous API). It's also a dependency that we may want to replace by something else later.
- having this logic related to the recent files here in ElectronMainMenu and also in MainFrame. Ideally both of them would just have functions to say "Hey this file was opened" and "Hey I want the list of recent files".

To do this, I suggest to look at `PreferencesContext`. It's a React context already used to get and save all the preferences of the app.
We could add two functions, one to get the recent files, one to declare a new file that is opened.
Then, we can use the context Consumer to pass to ElectronMainMenu and MainFrame the functions to get the recent files or declare a file.

If later the logic of the files or the preferences are becoming too complex, we can always split the preferences in multiple contexts... without changing the components. The components will only care about their props having the functions to get the recent files or declare an opened file :)